application/src/main/java/com/sanction/thunder/authentication/ThunderAuthenticator.java

```diff
@@ -0,0 +1,37 @@
+package com.sanction.thunder.authentication;
+
+import com.google.common.base.Optional;
+
+import io.dropwizard.auth.AuthenticationException;
+import io.dropwizard.auth.Authenticator;
+
+import io.dropwizard.auth.basic.BasicCredentials;
+
+public class ThunderAuthenticator implements Authenticator<BasicCredentials, Key> {
+
+  private String application;
+  private String secret;
+
+  public ThunderAuthenticator(String application, String secret) {
```

Comment:
Like I was saying above, maybe just take in a `List<Key>` here and save that instead of one application and one secret. That way we can add more.

And then in the authenticate method below, we just iterate through that list to check if its valid.